script/update-content-versions-and-check-for-archived-slugs.ts

```diff
@@ -0,0 +1,105 @@
+import fs from 'fs';
+import fetch from 'node-fetch';
+import path from 'path';
+import { argv } from 'yargs';
+import { LinkedArchiveTreeNode } from '../src/app/content/types';
+import { flattenArchiveTree } from '../src/app/content/utils';
+import archivedSlugs from '../src/archived-slugs.json';
+import { ARCHIVE_URL, REACT_APP_ARCHIVE_URL, REACT_APP_OS_WEB_API_URL } from '../src/config';
+import books from '../src/config.books.json';
+import createArchiveLoader from '../src/gateways/createArchiveLoader';
+import createOSWebLoader from '../src/gateways/createOSWebLoader';
+
+interface ArchivedSlug {
+  pathname: string;
+  bookId: string;
+  pageId: string;
+}
+
+type ArchivedSlugs = ArchivedSlug[];
+
+const { book, newVersion } = argv as any as {
+  book: string
+  newVersion: string | number;
+};
+
+(global as any).fetch = fetch;
+
+const booksPath = path.resolve(__dirname, '../src/config.books.json');
+const archivedSlugsPath = path.resolve(__dirname, '../src/archived-slugs.json');
+
+const archiveLoader = createArchiveLoader(`${ARCHIVE_URL}${REACT_APP_ARCHIVE_URL}`);
+const osWebLoader = createOSWebLoader(`${ARCHIVE_URL}${REACT_APP_OS_WEB_API_URL}`);
+
+async function processBooks() {
+  const booksData = Object.entries(books);
```

Comment:
why are you converting this to/from and entries list and doing all this searching for the index and stuff? wouldn't it be easier to just use it as an object keyed by the book id?